### PR TITLE
Initial working version of the formation parameterization

### PR DESCRIPTION
**Pull request by patrickherring-TRI**
This PR adds the ability to insert parameters for the formation template. Additional capabilities will be added as the project progresses.
- class method to insert the relevant parameters and check that the right step is being modified
- Tests to ensure that the correct method of cycle index assignment is being used

### CODE REVIEW

File: beep/generate_protocol.py [hunk 1]
```diff
@@ -48,9 +48,9 @@
 from monty.serialization import loadfn
 
 from beep import logger, __version__
-from beep.protocol import PROCEDURE_TEMPLATE_DIR
+from beep.protocol import PROCEDURE_TEMPLATE_DIR, BIOLOGIC_TEMPLATE_DIR
 from beep.protocol.maccor import Procedure
-
+from beep.protocol.biologic import Settings
```
Comment (chirranjeevigopal-TRI) [inline beep/generate_protocol.py:53]: Just to confirm, Biologic:Settings :: Maccor:Procedure :: Arbin:Schedule ?

File: beep/protocol/biologic.py [hunk 2]
```diff
@@ -164,3 +163,129 @@ def to_file(self, filename, encoding="ISO-8859-1", column_width=20, linesep="\r\
 
         with open(filename, "wb") as f:
             f.write(contents.encode(encoding))
+
+    def formation_protocol_bcs(self, params):
```
Comment (chirranjeevigopal-TRI) [inline beep/protocol/biologic.py:167]: is bcs an extension for one of biologic output files?

Comment (chirranjeevigopal-TRI) [reply beep/protocol/biologic.py:167]: boundary conditions? :-)

**Review by chirranjeevigopal-TRI — APPROVED** [review]
The methods look good to me, and very detailed. Thank you! I am not as familiar with the biologic settings, so I might have you walk me through some of the logic after merge.

File: beep/generate_protocol.py (hunk 1 above)
Comment (patrickherring-TRI) [reply beep/generate_protocol.py:53]: Yup!

File: beep/protocol/biologic.py (hunk 2 above)
Comment (patrickherring-TRI) [reply beep/protocol/biologic.py:167]: haha, I think that bcs is Biologic cycler system? Not sure actually :P

**Pull request merged**